### PR TITLE
Replace nc-vsock with ncat for vsock test

### PR DESCRIPTION
**Pull request by qcheng-redhat**
Replace nc-vsock with ncat for vsock test on rhel9

ID: 2034163

Signed-off-by: qcheng <qcheng@redhat.com>

### CODE REVIEW

**Comment by qcheng-redhat** [comment]
Test results:

 (1/5) Host_RHEL.m8.u6.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.6.0.x86_64.io-github-autotest-qemu.vsock_test.q35:PASS (77.59 s)
 (2/5) Host_RHEL.m8.u6.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.6.0.x86_64.io-github-autotest-qemu.vsock_test.negative_test.q35: PASS (188.25 s)
 (3/5) Host_RHEL.m8.u6.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.6.0.x86_64.io-github-autotest-qemu.vsock_test.hotplug.q35: PASS (160.74 s)
 (4/5) Host_RHEL.m8.u6.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.6.0.x86_64.io-github-autotest-qemu.vsock_test.migration_with_vsock.q35: PASS (246.44 s)
 (5/5) Host_RHEL.m8.u6.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.6.0.x86_64.io-github-autotest-qemu.vsock_test.iommu_enabled.q35: PASS (89.08 s)

 (1/5) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.vsock_test.q35: PASS (76.45 s)
 (2/5) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.vsock_test.negative_test.q35: PASS (257.56 s)
 (3/5) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.vsock_test.hotplug.q35: PASS (305.63 s)
 (4/5) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.vsock_test.migration_with_vsock.q35: PASS (235.24 s)
 (5/5) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.vsock_test.iommu_enabled.q35: PASS (77.30 s)

**Comment by qcheng-redhat** [comment]
Hi, @PaulYuuu @zhenyzha Please help review this patch when you have time.

**Comment by PaulYuuu** [comment]
Can we use `nc` in all products?

**Comment by qcheng-redhat** [comment]
> Can we use `nc` in all products?
No. Both nc and ncat on rhel 8 are 7.70, which don't support vsock. Since nc/ncat 7.8, introduced vsock support. On rhel 9, nc/ncat are 7.91.

**Comment by zhenyzha** [comment]
```
 (1/5) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.aarch64.io-github-autotest-qemu.vsock_test.arm64-pci: PASS (283.23 s)
 (2/5) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.aarch64.io-github-autotest-qemu.vsock_test.negative_test.arm64-pci: FAIL: vsock listening prcoess inside guest does not exit after close host nc-vsock connection. (398.10 s)
 (3/5) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.aarch64.io-github-autotest-qemu.vsock_test.hotplug.arm64-pci: FAIL: vsock listening prcoess inside guest does not exit after close host nc-vsock connection. (408.92 s)
 (4/5) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.aarch64.io-github-autotest-qemu.vsock_test.migration_with_vsock.arm64-pci: PASS (532.43 s)
 (5/5) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.aarch64.io-github-autotest-qemu.vsock_test.iommu_enabled.arm64-pci: PASS (222.41 s)
```

Need further analysis, log upload to the bug.

**Comment by qcheng-redhat** [comment]
> ```
>  (1/5) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.aarch64.io-github-autotest-qemu.vsock_test.arm64-pci: PASS (283.23 s)
>  (2/5) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.aarch64.io-github-autotest-qemu.vsock_test.negative_test.arm64-pci: FAIL: vsock listening prcoess inside guest does not exit after close host nc-vsock connection. (398.10 s)
>  (3/5) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.aarch64.io-github-autotest-qemu.vsock_test.hotplug.arm64-pci: FAIL: vsock listening prcoess inside guest does not exit after close host nc-vsock connection. (408.92 s)
>  (4/5) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.aarch64.io-github-autotest-qemu.vsock_test.migration_with_vsock.arm64-pci: PASS (532.43 s)
>  (5/5) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.aarch64.io-github-autotest-qemu.vsock_test.iommu_enabled.arm64-pci: PASS (222.41 s)
> ```
> 
> Need further analysis, log upload to the bug.

The root cause is that when terminate the vsock connection from host, the vsock connection in guest will also be  terminated, but it takes much longer time on arm than x86 platform, especially for large file transferring.  Now it is fixed. 
Please have a try again. Thanks. @zhenyzha

**Comment by zhenyzha** [comment]
Power:
```
 (1/5) Host_RHEL.m8.u6.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.6.0.ppc64le.io-github-autotest-qemu.vsock_test: PASS (415.35 s)
 (2/5) Host_RHEL.m8.u6.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.6.0.ppc64le.io-github-autotest-qemu.vsock_test.negative_test: PASS (293.58 s)
 (3/5) Host_RHEL.m8.u6.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.6.0.ppc64le.io-github-autotest-qemu.vsock_test.hotplug: PASS (336.77 s)
 (4/5) Host_RHEL.m8.u6.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.6.0.ppc64le.io-github-autotest-qemu.vsock_test.migration_with_vsock: PASS (398.90 s)
 (5/5) Host_RHEL.m8.u6.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.6.0.ppc64le.io-github-autotest-qemu.vsock_test.iommu_enabled: PASS (107.41 s)

 (1/5) Host_RHEL.m8.u6.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.ppc64le.io-github-autotest-qemu.vsock_test: PASS (88.55 s)
 (2/5) Host_RHEL.m8.u6.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.ppc64le.io-github-autotest-qemu.vsock_test.negative_test: PASS (240.11 s)
 (3/5) Host_RHEL.m8.u6.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.ppc64le.io-github-autotest-qemu.vsock_test.hotplug: PASS (282.78 s)
 (4/5) Host_RHEL.m8.u6.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.ppc64le.io-github-autotest-qemu.vsock_test.migration_with_vsock: PASS (316.45 s)
 (5/5) Host_RHEL.m8.u6.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.ppc64le.io-github-autotest-qemu.vsock_test.iommu_enabled: PASS (97.24 s)
```


Aarch64:
```
 (1/6) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.6.0.aarch64.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.default_install.aio_threads.arm64-pci: PASS (1001.00 s)
 (2/6) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.6.0.aarch64.io-github-autotest-qemu.vsock_test.arm64-pci: PASS (79.19 s)
 (3/6) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.6.0.aarch64.io-github-autotest-qemu.vsock_test.negative_test.arm64-pci: PASS (101.63 s)
 (4/6) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.6.0.aarch64.io-github-autotest-qemu.vsock_test.hotplug.arm64-pci: PASS (162.89 s)
 (5/6) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.6.0.aarch64.io-github-autotest-qemu.vsock_test.migration_with_vsock.arm64-pci: PASS (307.10 s)
 (6/6) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.6.0.aarch64.io-github-autotest-qemu.vsock_test.iommu_enabled.arm64-pci: PASS (70.68 s)

 (1/5) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.aarch64.io-github-autotest-qemu.vsock_test.arm64-pci: PASS (100.21 s)
 (2/5) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.aarch64.io-github-autotest-qemu.vsock_test.negative_test.arm64-pci: PASS (114.47 s)
 (3/5) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.aarch64.io-github-autotest-qemu.vsock_test.hotplug.arm64-pci: PASS (181.11 s)
 (4/5) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.aarch64.io-github-autotest-qemu.vsock_test.migration_with_vsock.arm64-pci: PASS (340.05 s)   
 (5/5) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.aarch64.io-github-autotest-qemu.vsock_test.iommu_enabled.arm64-pci: PASS (94.04 s)
```
LGTM.Acked-by: Zhenyu Zhang <zhenyzha@redhat.com>

**Review by zhenyzha — APPROVED** [review]

**Comment by XueqiangWei** [comment]
@PaulYuuu Please help review it. Many thanks.

File: qemu/tests/migration_with_vsock.py [hunk 1]
```diff
@@ -61,19 +61,27 @@ def ping_pong_migration(repeat_times):
...
         return host_vsock_session
 
+    vsock_test_tool = params["vsock_test_tool"]
     vm = env.get_vm(params["main_vm"])
     session = vm.wait_for_login()
-    nc_vsock_bin = compile_nc_vsock(test, vm, session)
+    if vsock_test_tool == "nc_vsock":
+        tool_bin = compile_nc_vsock(test, vm, session)
+    if vsock_test_tool == "ncat":
+        tool_bin = "/usr/bin/ncat"
```
Comment (PaulYuuu) [inline qemu/tests/migration_with_vsock.py:84]: Suggest to use `path.find_command("ncat")` to get the real path.

File: qemu/tests/vsock_hotplug.py [hunk 2]
```diff
@@ -74,14 +77,17 @@ def run(test, params, env):
                     test.fail("dmesg check failed: %s" % error_msg)
     # Transfer data from guest to host
     try:
-        nc_vsock_bin = vsock_test.compile_nc_vsock(test, vm, session)
+        if vsock_test_tool == "nc_vsock":
+            tool_bin = vsock_test.compile_nc_vsock(test, vm, session)
+        if vsock_test_tool == "ncat":
+            tool_bin = "/usr/bin/ncat"
```
Comment (PaulYuuu) [inline qemu/tests/vsock_hotplug.py:83]: Same above.

File: qemu/tests/migration_with_vsock.py (hunk 1 above)
Comment (qcheng-redhat) [reply qemu/tests/migration_with_vsock.py]: Updated.

File: qemu/tests/vsock_hotplug.py (hunk 2 above)
Comment (qcheng-redhat) [reply qemu/tests/vsock_hotplug.py]: Updated.

**Comment by PaulYuuu** [comment]
@qcheng-redhat you should not add the `ID` in the commit meesage. cc @XueqiangWei

**Comment by qcheng-redhat** [comment]
> 

Updated. 

> @qcheng-redhat you should not add the `ID` in the commit meesage. cc @XueqiangWei

Updated.

**Comment by zhenyzha** [comment]
No regression

```
 (1/5) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.aarch64.io-github-autotest-qemu.vsock_test.arm64-pci: PASS (93.24 s)
 (2/5) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.aarch64.io-github-autotest-qemu.vsock_test.negative_test.arm64-pci: PASS (73.49 s)
 (3/5) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.aarch64.io-github-autotest-qemu.vsock_test.hotplug.arm64-pci: PASS (141.20 s)
 (4/5) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.aarch64.io-github-autotest-qemu.vsock_test.migration_with_vsock.arm64-pci: PASS (249.58 s)
 (5/5) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.aarch64.io-github-autotest-qemu.vsock_test.iommu_enabled.arm64-pci: PASS (94.50 s)
```

**Comment by qcheng-redhat** [comment]
x86:
 (1/5) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.vsock_test.q35: PASS (79.17 s)
 (2/5) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.vsock_test.negative_test.q35: PASS (255.10 s)
 (3/5) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.vsock_test.hotplug.q35:PASS (297.13 s)
 (4/5) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.vsock_test.migration_with_vsock.q35: PASS (229.55 s)
 (5/5) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.vsock_test.iommu_enabled.q35: PASS (79.17 s)

**Review by PaulYuuu — APPROVED** [review]

**Pull request merged**